latest/index.bs

```diff
@@ -331,24 +317,1002 @@ Conforming readers:
+highlight: json
+</pre>
+
+describes the function
+
+```
+x = (i + 0.1) * 2
+y = (j + 0.9) * 3
+```
+
+and is invertible.
+</div>
+
+
+#### coordinates and displacements
+
+`coordinates` and `displacements` transformations store coordinates or displacements in an array and interpret them as a transformation.
+Applying the transformation amounts to looking up the appropriate locations in the array and interpolating.
+`coordinates` and `displacements` transformations are not invertible in general, but implementations
+MAY approximate their inverses. Metadata for these coordinate transforms have the following field: 
+
+<dl>
+  <dt><strong>path</strong></dt>
+  <dd>  The location of the coordinate array in this (or another) containter.</dd>
+  <dt><strong>interpolation</strong></dt>
+  <dd>  The `interpolation` attributes MAY be provided. It's value indicates
+        the interpolation to use if transforing points not on the array's discrete grid.
+        Values could be:
+        <ul>
+            <li><code>linear</code> (default)</li>
+            <li><code>nearest</code></li>
+            <li><code>cubic</code></li>
+        </ul></dd>
+</dl>
+
+The array data at `path` MUST define space and coordinate transform metadata:
+
+* space metadata MUST have exactly one axis with `"type" : "coordinate"`
+* Every axis name in the `coordinateTransform`'s `input` MUST appear in the space
+* The array dimension corresponding to the coordinate axis MUST have length greater than or equal to the dimensionality of the `coordinateTransform` `output`
+* SHOULD have `name` equal to the `name` of the corresponding `coordinateTransform`.
+
+For `coordinates`:
+
+* If a `coordinates`' input space has dimensionality `N`, then the array data at `path` MUST have dimensionality equal to `(N + 1)`.
+* coordinateSystem metadata MUST have exactly one axis with `"type" : "coordinate"`
+
+For `displacements`:
+* coordinateSystem metadata MUST have exactly one axis with `"type" : "displacement"`
+* `input` and `output` MUST have an equal number of dimensions.
+
+For example, in 1D:
+```
+{
+    "name" : "a coordinate field transform",
+    "type": "coordinates",
+    "path" : "i2xCoordinates",
+    "input" : "i",
+    "output" : "x",
+    "interpolation" : "nearest"
+}
+```
+
+where we assume input spaces "i" and "x" are defined elsewhere.
+Example metadata for the array data at path `coordinates` above:
+
+```
+{
+    "coordinateSystems" : [
+        {
+            "name" : "a coordinate field transform",
+            "axes" : [
+                { "name": "i", "type": "space", "discrete": true },
+                { "name": "c", "type": "coordinate", "discrete": true }
+            ]
+        } 
+    ],
+    "coordinateTransformations" : [
+        {
+            "type" : "identity",
+            "output" : "a coordinate field transform"
+        }
+    ]
+}
+```
+
+If the arrray in `coordinates` contains the data: `[-9, 9, 0]`, then this metadata defines the function:
```

Comment:
this is confusing, what are `i` and `x` here? Seems much simpler to say that the array defines data, which can be resampled in various ways (e.g., nearest neighbor interpolation, which this function implements)

Reply:
`i` is [defined here](https://github.com/bogovicj/ngff/blob/coord-transforms/latest/index.bs#L1151), and `x` is [defined here](https://github.com/bogovicj/ngff/blob/coord-transforms/latest/index.bs#L1137).
I.e. the transformation is a function `x = f(i)`.

I wanted to include a concrete example.